datadog/fwprovider/data_source_datadog_integration_aws_external_id.go

```diff
@@ -0,0 +1,160 @@
+	"github.com/hashicorp/terraform-plugin-framework/types"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+var (
+	_ datasource.DataSource = &awsIntegrationExternalIDDataSource{}
+)
+
+func NewAwsIntegrationExternalIDDataSource() datasource.DataSource {
+	return &awsIntegrationExternalIDDataSource{}
+}
+
+type awsIntegrationExternalIDDataSourceModel struct {
+	ID           types.String `tfsdk:"id"`
+	AwsAccountId types.String `tfsdk:"aws_account_id"`
+	ExternalId   types.String `tfsdk:"external_id"`
+	AwsPartition types.String `tfsdk:"aws_partition"`
+}
+
+type awsIntegrationExternalIDDataSource struct {
+	Api  *datadogV2.AWSIntegrationApi
+	Auth context.Context
+}
+
+func (d *awsIntegrationExternalIDDataSource) Configure(_ context.Context, request datasource.ConfigureRequest, response *datasource.ConfigureResponse) {
+	providerData, _ := request.ProviderData.(*FrameworkProvider)
+	d.Api = providerData.DatadogApiInstances.GetAWSIntegrationApiV2()
+	d.Auth = providerData.Auth
+}
+
+func (d *awsIntegrationExternalIDDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = "integration_aws_external_id"
+}
+
+func (d *awsIntegrationExternalIDDataSource) Schema(_ context.Context, _ datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Use this data source to retrieve the external ID from an existing AWS integration. This can be used to reference the external ID value from an existing AWS account integration.",
+		Attributes: map[string]schema.Attribute{
+			// Datasource ID
+			"id": utils.ResourceIDAttribute(),
+			"aws_account_id": schema.StringAttribute{
+				Description: "The AWS account ID of the integration to retrieve the external ID from.",
+				Required:    true,
+				Validators: []validator.String{
+					stringvalidator.RegexMatches(regexp.MustCompile(`^[0-9]{12}$`), "invalid aws_account_id"),
+				},
+			},
+			"external_id": schema.StringAttribute{
+				Description: "The external ID associated with the AWS integration.",
+				Computed:    true,
+			},
+			"aws_partition": schema.StringAttribute{
+				Description: "Optional AWS account partition to disambiguate when multiple integrations exist for the same account ID.",
+				Optional:    true,
+				Validators: []validator.String{
+					stringvalidator.OneOf("aws", "aws-cn", "aws-us-gov"),
+				},
+			},
+		},
+	}
+}
+
+func (d *awsIntegrationExternalIDDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var state awsIntegrationExternalIDDataSourceModel
+	resp.Diagnostics.Append(req.Config.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	awsAccountId := state.AwsAccountId.ValueString()
+
+	// Retrieve AWS account integrations and filter to the requested account ID (and partition, if provided)
+	accountsResp, httpResp, err := d.Api.ListAWSAccounts(d.Auth)
```

Comment:
💭 : You can filter by `AwsAccountId` in the API call - https://github.com/DataDog/datadog-api-client-go/blob/83925bde768a179ecc055a696e2e4d803ddfbeab/api/datadogV2/api_aws_integration.go#L554